### PR TITLE
Sanitize quoted subject lines

### PR DESCRIPTION
**Pull request by ibaned**
Fixes compile errors when most recent commit message contains double quotes

### CODE REVIEW

**Comment by ye-luo** [comment]
okay to test

**Comment by markdewing** [comment]
Unfortunately there is a situation where this code results in incorrect output in git-rev.h

EXECUTE_PROCESS runs at configuration time, custom commands run at compile time.  This code only captures the git subject line at configuration time.

The steps:
1. Commit a source change
2. Run cmake
3. Build
4. Make a change in the code and check into git
5. Build - assuming the change in 4 did not force a reconfiguration - now the subject line will be incorrect, and will not match the git revision
    The subject line will be that of the commit from 1., but the git revision will be from the commit in 4.


Possible solutions:
1. Assume 'sed' exists on target machine and use that to escape quotes as a step in the custom command.
2. Use the %f format to git log, which produces sanitized output suitable for filenames.  Spaces are replaced with dashes, which makes it a bit ugly.
3. Decide that capturing the subject line from the most recent commit is more trouble than it's worth, and only collect the hash.
4. Leave the code as-is and require developers to amend the subject line in the most recent commit, or add another commit with a safe subject.
5. Something else?

**Comment by ibaned** [comment]
Out of (1)-(4), I'm leaning towards (1).

**Comment by markdewing** [comment]
If you want to implement the 'sed' solution, please go ahead (otherwise I can do it).  It would be good to have a solution for this problem.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -50,9 +51,9 @@ IF (GIT_FOUND AND IS_GIT_PROJECT)
     COMMAND ${CMAKE_COMMAND} -E echo_append "#define GIT_COMMIT_LAST_CHANGED_RAW " >> ${GITREV_TMP}
     COMMAND ${GIT_EXECUTABLE} log -1 --format=%ad >> ${GITREV_TMP}
     COMMAND ${CMAKE_COMMAND} -E echo >> ${GITREV_TMP}
-    COMMAND ${CMAKE_COMMAND} -E echo_append "#define GIT_COMMIT_SUBJECT_RAW " >> ${GITREV_TMP}
-    COMMAND ${GIT_EXECUTABLE} log -1 --format="%s" >> ${GITREV_TMP}
-    COMMAND ${CMAKE_COMMAND} -E echo >> ${GITREV_TMP}
+    COMMAND ${CMAKE_COMMAND} -E echo_append "#define GIT_COMMIT_SUBJECT_RAW \"" >> ${GITREV_TMP}
+    COMMAND ${GIT_EXECUTABLE} log -1 --format=%s | sed -E "s/\"/\\\"/g" | tr -d '\\n' >> ${GITREV_TMP}
```
Comment (markdewing) [inline src/CMakeLists.txt]: To make this work my Ubuntu 14.04 box, I need to add a couple more backslashes to the sed replacement:  sed -E "s/\"/\\\\\"/g"

**Comment by grahamlopez** [comment]
When this PR is settled and merged, I will tag a new release. Since this change is relatively simple, it would be good if it were today or tomorrow if possible.

@markdewing I'm going to leave the final review/merge to you.

**Comment by ibaned** [comment]
I've added the missing slashes pointed out by @markdewing

**Pull request merged**